[Hugo-Chia] iP

Hugo-Chia opened the pull request:
Create a PR to the upstream repo

### **ArtemisPro**

> "One small step for man, one giant leap for mankind" - Neil Armstrong ([source](https://www.nasa.gov/history/50-years-ago-one-small-step-one-giant-leap/))

ArtemisPro helps you remember tasks so we can _get back to the moon again_. It's

- text-based
- lightweight
- ~~easy~~ VERY EASY to learn and use

All you need to do is,

1. Download from it [here](https://github.com/Hugo-Chia/ip/releases/tag/A-Jar)
2. Double-click it
3. Then you can use it to manage yours task. It's this easy! 😱

The best part is this is **FREE**!

Features:

- [X] Manage tasks
- [ ] Manage deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```java
public static void main(String[] args) {
        new Artemis("artemis.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Hugo-Chia committed:
Implemented Level-0: Rename, Greet and Exit

Hugo-Chia committed:
Added echo feature

Hugo-Chia committed:
Added a list feature for the chatbot

Hugo-Chia committed:
Added ability to mark tasks as done/not done

Hugo-Chia committed:
Added ToDos, Events, Deadlines

Hugo-Chia committed:
Fixed buggy code in event matching by replacing substring with startsWith

Hugo-Chia committed:
Added basic error handling

Hugo-Chia committed:
Added Automated Text UI Testing

Hugo-Chia committed:
Added basic error handling for todo and generic errors

Hugo-Chia committed:
Implemented delete tasks feature

Hugo-Chia committed:
Added Enum Commands

Hugo-Chia committed:
Improved error handling

Hugo-Chia committed:
Updated Automated Text UI Testing

Hugo-Chia committed:
Add data file creation logic

Hugo-Chia committed:
Add save data feature

User can now save data.

Previously, the chatbot is memoryless. Now, it is more useful as the
state of data is saved.

Hugo-Chia committed:
Merge branch 'branch-Level-7'

Hugo-Chia committed:
Modify ENUM to follow Java conventions

Hugo-Chia committed:
Add date-time support for Deadline Task

Hugo-Chia committed:
Add date-time support for Event Task

Hugo-Chia committed:
Merge branch 'branch-Level-8'

Hugo-Chia committed:
Move data code from Main class to Storage class

Hugo-Chia committed:
Fix Text UI Test

Hugo-Chia committed:
Fix pathname bug

Hugo-Chia committed:
Merge branch 'Text-UI-Fix'

Hugo-Chia committed:
Merge branch 'master' into A-MoreOOP

Hugo-Chia committed:
Add TaskList class

Hugo-Chia committed:
Add Ui class

Hugo-Chia committed:
Add Parser class

Hugo-Chia committed:
Update test case - EXPECTED.TXT

Hugo-Chia committed:
Update Parser

Hugo-Chia committed:
Update test case - EXPECTED.txt

Hugo-Chia committed:
Refactor code into packages

Hugo-Chia committed:
Merge branch 'A-Packages'

Hugo-Chia committed:
Merge branch 'add-gradle-support' of https://github.com/Hugo-Chia/ip

Hugo-Chia committed:
Add JUnit test for Parser and Deadline

Hugo-Chia committed:
Merge tag 'A-JUnit'

Hugo-Chia committed:
Add Javadoc comments for Ui, Command and Storage packages

Hugo-Chia committed:
Add Javadoc comments for Task package

Hugo-Chia committed:
Follow Coding Standard

Hugo-Chia committed:
Add find feature

Hugo-Chia committed:
Merge branch 'branch-A-JavaDoc'

Hugo-Chia committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/artemis/command/Parser.java

Hugo-Chia committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/artemis/command/Parser.java
#	src/main/java/artemis/task/TaskList.java
#	src/main/java/artemis/ui/Ui.java

Hugo-Chia committed:
Remove unused code in Deadline and Event classes

Hugo-Chia committed:
Refactor code to include Command class and subclasses

Hugo-Chia committed:
Add CheckStyle

Hugo-Chia committed:
Merge tag 'A-CheckStyle'

Hugo-Chia committed:
Merge branch 'master' into Command-Refactor

aisyahally commented on src/main/java/artemis/ui/Artemis.java:
```diff
@@ -0,0 +1,157 @@
+import artemis.task.Todo;
+
+
+public class Artemis {
```
It seems like the Artemis class is handling everything. Might be better if the function was broken down into more parts (and allocated to other classes) for better readability and debugging.

Hugo-Chia replied:
I have fixed the code in commit: 36aa3eb5488a1e3e37dd3eb717984017ffc9992f and merged into the master branch. I appreciate the feedback.

You may find the relevant code is in the [Command-Refactor](https://github.com/Hugo-Chia/ip/tree/Command-Refactor) branch.

aisyahally commented on src/main/java/artemis/command/Parser.java:
```diff
@@ -0,0 +1,82 @@
+    /**
+     * Creates an instance of the Parser object.
+     */
+    public Parser() {
```
Empty constructor may be redundant

aisyahally commented on src/main/java/artemis/command/ArtemisException.java:
```diff
@@ -0,0 +1,7 @@
+package artemis.command;
+
+public class ArtemisException extends Exception {
```
Could consider creating different exceptions to handle errors differently rather than having just a general one

aisyahally reviewed (commented):
Overall the code looks great! There were no obvious coding standard violations and it was generally readable. Good job :)

Hugo-Chia committed:
Add Unmark Command

Hugo-Chia committed:
Add more Command subclasses

Hugo-Chia committed:
Merge branch 'Command-Refactor'

Hugo-Chia committed:
Merge branch 'master' into branch-Level-10

Hugo-Chia committed:
Add JavaFX UI to Artemis

Hugo-Chia committed:
Merge branch 'branch-Level-10'

kerktaiheng commented on src/main/java/artemis/ui/Artemis.java:
```diff
@@ -36,6 +36,15 @@ public void run() {
+                    try {
+                        keyword = Parser.parseKeywordCommand(userInput);
+                    } catch (ArrayIndexOutOfBoundsException e) {
+                        throw new ArtemisException("You did not choose a valid task to mark. Please try again!!! :(\n");
```
In general, it is a good idea to have constant strings in its own file so that you are guaranteed consistency and you can make convenient changes when the component is refactored

kerktaiheng commented on src/main/java/artemis/ui/Ui.java:
```diff
@@ -29,6 +29,19 @@ public void listTask(TaskList taskList) {
+     * @param taskList A filtered tasklist.
+     */
+    public void listMatchingTask(TaskList taskList) {
+        System.out.println("Here are the matching tasks in your list:\n");
```
Same as above

kerktaiheng commented on src/main/java/Artemis.java:
```diff
@@ -46,16 +42,12 @@ public void run() {
-                        throw new ArtemisException("You did not choose a task to unmark. Please try again!!! :(\n");
-                    }
-
+                } else if (command.equals(Commands.UNMARK.name().toLowerCase())) {
```
this defeats the purpose of the enum 
but thats javas fault for being lame

kerktaiheng commented on src/main/java/Storage.java:
```diff
@@ -4,19 +4,63 @@
+                //e.printStackTrace();
+                System.out.println(e.getMessage());
+            }
+
```
remove commented code

kerktaiheng reviewed (commented):
Yapping for participation marks as always

Frankly not much to say at all

Great stuff!

Hugo-Chia committed:
Refactor UI related classes to UI package

Move UI related classes to the UI package. This helps to improve code
quality.

Hugo-Chia committed:
Fix bug in build.gradle

Modified the mainClass to the correct package name for Launcher

Hugo-Chia committed:
Fix UI Bug in Ui.java

Fix a text bug where \\n is displayed in the GUI output.

Hugo-Chia committed:
Merge branch 'master' into branch-A-Assertions

Hugo-Chia committed:
Merge branch 'master' into branch-A-CodeQuality

Hugo-Chia committed:
Add Assertion

Ensures no response to the user will be null.

Hugo-Chia committed:
Remove duplicated strings in MarkCommand and UnmarkCommand

Hugo-Chia committed:
Remove unused comments

Hugo-Chia committed:
Remove magic literals

Hugo-Chia committed:
Remove duplicated String

Remove magic literals as it is not good practice.

Remove duplicated String and magic literals by creating a new exception
class - ArtemisDateTimeException.

This will align the code with best practices.

Hugo-Chia committed:
Cleanup code

Removed duplicate strings and magic literals to improve code quality.

Hugo-Chia committed:
Merge pull request #1 from Hugo-Chia/branch-A-Assertions

Branch-A-Assertions

Hugo-Chia committed:
Merge branch 'master' into branch-A-CodeQuality

Hugo-Chia committed:
Merge pull request #2 from Hugo-Chia/branch-A-CodeQuality

Branch-A-CodeQuality

Hugo-Chia committed:
Improve Code Quality

Refactor code by creating new methods and removing multiple layers of
abstraction.

Hugo-Chia committed:
Add sort feature

The sort feature allows user to sort by task name or task date. This
improves usability of the app.

Hugo-Chia committed:
Merge tag 'BCD-Extension'

Hugo-Chia committed:
Add gradle.yml to set up CI

Hugo-Chia committed:
Update Readme.md

Hugo-Chia committed:
Add dynamic resizing feature

Hugo-Chia committed:
Enhance UI

To provide a better user experience:

1. Change background color to light blue
2. Change profile picture to be circular

Hugo-Chia committed:
Merge branch 'A-BetterGui'

Hugo-Chia committed:
Merge branch 'master' into UserGuide

Hugo-Chia committed:
Add Partial Readme

Hugo-Chia committed:
Merge branch 'UserGuide'

Hugo-Chia committed:
Add more information to User Guide

Hugo-Chia committed:
Merge branch 'UserGuide'
